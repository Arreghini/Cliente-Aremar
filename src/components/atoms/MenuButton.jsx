import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars, faTimes } from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";

const MenuButton = ({ isOpen, onClick }) => (
  <button
    onClick={onClick}
    className="text-3xl md:text-4xl text-white hover:text-yellow-400 focus:outline-none"
  >
    <FontAwesomeIcon icon={isOpen ? faTimes : faBars} />
  </button>
);
MenuButton.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MenuButton;
