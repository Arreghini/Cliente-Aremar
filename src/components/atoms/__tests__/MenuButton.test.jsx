// src/components/atoms/__tests__/MenuButton.test.jsx
import React from 'react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import MenuButton from '../MenuButton';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons';

// Mock del FontAwesomeIcon para simplificar las pruebas
vi.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: ({ icon }) => <span data-testid="icon">{icon.iconName}</span>,
}));

describe('MenuButton', () => {
  let onClickMock;

  beforeEach(() => {
    onClickMock = vi.fn();
  });

  it('renderiza el botón correctamente', () => {
    render(<MenuButton isOpen={false} onClick={onClickMock} />);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
  });

  it('muestra el ícono faBars cuando isOpen es false', () => {
    render(<MenuButton isOpen={false} onClick={onClickMock} />);
    const icon = screen.getByTestId('icon');
    expect(icon).toHaveTextContent('bars');
  });

  it('muestra el ícono faTimes cuando isOpen es true', () => {
  render(<MenuButton isOpen={true} onClick={onClickMock} />);
  const icon = screen.getByTestId('icon');
  expect(icon).toHaveTextContent('xmark'); // <-- cambio de 'times' a 'xmark'
});

  it('llama a onClick al hacer clic', () => {
    render(<MenuButton isOpen={false} onClick={onClickMock} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('coincide con el snapshot', () => {
    const { container } = render(<MenuButton isOpen={false} onClick={onClickMock} />);
    expect(container).toMatchSnapshot();
  });
});
