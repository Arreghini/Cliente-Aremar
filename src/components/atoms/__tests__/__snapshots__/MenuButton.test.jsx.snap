// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuButton > coincide con el snapshot 1`] = `
<div>
  <button
    class="text-3xl md:text-4xl text-white hover:text-yellow-400 focus:outline-none"
  >
    <span
      data-testid="icon"
    >
      bars
    </span>
  </button>
</div>
`;
